### PR TITLE
fix!: Provide primary keys for tables (D)

### PR DESCRIPTION
**Pull request by candiduslynx**
Part of https://github.com/cloudquery/cloudquery/issues/15381

BEGIN_COMMIT_OVERRIDE
fix: Provide primary keys for tables (C) (https://github.com/cloudquery/cloudquery/pull/15415)
BREAKING-CHANGE: Provide primary keys for tables (C) (https://github.com/cloudquery/cloudquery/pull/15415). The following tables are affected:
* `aws_directconnect_gateway_associations`
* `aws_directconnect_gateway_attachments`
* `aws_docdb_cluster_parameters`
* `aws_docdb_event_categories`
* `aws_docdb_event_subscriptions`
* `aws_docdb_events`
* `aws_docdb_orderable_db_instance_options`
* `aws_docdb_pending_maintenance_actions`
* `aws_dynamodb_table_replica_auto_scalings`

END_COMMIT_OVERRIDE

### CODE REVIEW

File: plugins/source/aws/resources/services/docdb/cluster_parameters.go [hunk 1]
```diff
@@ -20,46 +19,29 @@ func clusterParameters() *schema.Table {
 		Resolver:    fetchDocdbClusterParameters,
 		Transform:   transformers.TransformWithStruct(&types.Parameter{}),
```
Comment (disq) [inline plugins/source/aws/resources/services/docdb/cluster_parameters.go:20]:
```suggestion
		Transform:   transformers.TransformWithStruct(&types.Parameter{}, transformers.WithPrimaryKeys("ParameterName")),
```

File: plugins/source/aws/resources/services/docdb/cluster_parameters.go [hunk 2]
```diff
@@ -20,46 +19,29 @@ func clusterParameters() *schema.Table {
...
 		Columns: []schema.Column{
-			client.DefaultAccountIDColumn(false),
-			client.DefaultRegionColumn(false),
+			client.DefaultAccountIDColumn(true),
+			client.DefaultRegionColumn(true),
+			{
```
Comment (disq) [inline plugins/source/aws/resources/services/docdb/cluster_parameters.go:24]: Why not add `arn` to parent just like we have in rds/neptune?

File: plugins/source/aws/resources/services/docdb/cluster_parameters.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/docdb/cluster_parameters.go]: + 308aa898ac99e41060c46eb3876d552a59c94fcd

File: plugins/source/aws/resources/services/docdb/cluster_parameters.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/docdb/cluster_parameters.go:24]: The parent here is engine version that doesn't have ARN, as defined by the [docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazondocumentdbelasticclusters.html#amazondocumentdbelasticclusters-resources-for-iam-policies)

File: plugins/source/aws/docs/tables/aws_docdb_events.md [hunk 3]
```diff
@@ -4,19 +4,19 @@ This table shows data for Amazon DocumentDB Events.
 
 https://docs.aws.amazon.com/documentdb/latest/developerguide/API_Event.html
 
-The primary key for this table is **_cq_id**.
+The composite primary key for this table is (**account_id**, **region**, **date**, **source_arn**, **source_identifier**).
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/source/aws/docs/tables/aws_docdb_events.md]: I've added a concatenation of the event categories to the PK in e4c0ccf74436eed35845ff10e6e958a773729cb3

**Review by disq — APPROVED** [review]

**Review by bbernays — APPROVED** [review]

**Pull request merged**